sqlglot/expressions.py

```diff
@@ -3778,24 +3778,37 @@ class Type(AutoName):
     TEXT_TYPES = {
         Type.CHAR,
         Type.NCHAR,
-        Type.VARCHAR,
         Type.NVARCHAR,
         Type.TEXT,
+        Type.VARCHAR,
     }
 
     INTEGER_TYPES = {
-        Type.INT,
-        Type.TINYINT,
-        Type.SMALLINT,
         Type.BIGINT,
+        Type.BIT,
+        Type.INT,
         Type.INT128,
         Type.INT256,
-        Type.BIT,
+        Type.MEDIUMINT,
+        Type.SMALLINT,
+        Type.TINYINT,
+        Type.UBIGINT,
+        Type.UINT,
+        Type.UINT128,
+        Type.UINT256,
+        Type.UMEDIUMINT,
+        Type.USMALLINT,
+        Type.UTINYINT,
     }
 
     FLOAT_TYPES = {
-        Type.FLOAT,
+        Type.BIGDECIMAL,
```

Comment:
I think DECIMAL et al. are definitively _not_ floating point, right?

Reply:
Do we want something like `REAL_TYPES`?

Reply:
Donezo - https://github.com/tobymao/sqlglot/pull/3094/commits/18f513c413287023fe059e4ec1ba255a46294eaf